lib/http/request.rb

```diff
@@ -80,6 +82,8 @@ def initialize(verb, uri, headers = {}, proxy = {}, body = nil, version = "1.1")
 
       @headers["Host"]        ||= default_host
       @headers["User-Agent"]  ||= USER_AGENT
+      now = Time.now
+      @headers["Date"] = now.httpdate
```

Comment:
> Clients SHOULD only send a Date header field in messages that include an entity-body, as in the case of the PUT and POST requests, and even then it is optional. A client without a clock MUST NOT send a Date header field in a request.

> The HTTP-date sent in a Date header SHOULD NOT represent a date and time subsequent to the generation of the message. It SHOULD represent the best available approximation of the date and time of message generation, unless the implementation has no means of generating a reasonably accurate date and time. In theory, the date ought to represent the moment just before the entity is generated. In practice, the date can be generated at any time during the message origination without affecting its semantic value. 

http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.18

So, looks like this should go to `Request::Caching` instead. And also we MUST not override headers if they were explicitly given by user.